src/ucp/wireup/wireup.c

```diff
@@ -1372,43 +1512,34 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key,
         ucs_assert(ucp_ep_get_lane(ep, lane) != NULL);
     }
 
-    cm_wireup_ep = ucp_ep_get_cm_wireup_ep(ep);
-    ucs_assert(cm_wireup_ep != NULL);
-
     old_key = &ucp_ep_config(ep)->key;
     ucp_ep_config_lanes_intersect(old_key, new_key, ep, remote_address,
                                   addr_indices, reuse_lane_map);
 
-    /* CM lane has to be re-used by the new EP configuration */
-    ucs_assert(reuse_lane_map[ucp_ep_get_cm_lane(ep)] != UCP_NULL_LANE);
+    if (ucp_ep_has_cm_lane(ep)) {
+        /* CM lane has to be re-used by the new EP configuration */
+        ucs_assert(reuse_lane_map[ucp_ep_get_cm_lane(ep)] != UCP_NULL_LANE);
+        /* wireup lane hasn't been selected by the new configuration: only this
+         * function should select it */
+        ucs_assert(new_key->wireup_msg_lane == UCP_NULL_LANE);
```

Comment:
```suggestion
        ucs_assertv(new_key->wireup_msg_lane == UCP_NULL_LANE, "new_key->wireup_msg_lane=%u",
                    new_key->wireup_msg_lane);
```

Reply:
that's code from master. 
Was marked as "new added" code for some wrong reason.

Reply:
This code is indeed from master. Marked as diff because of indentation (`if` block).
IMO the suggestions makes sense.